exec/cnex/nstring.c

```diff
@@ -145,6 +179,17 @@ void string_resizeString(TString *s, size_t n)
     s->length = n;
 }
 
+TString *string_reverseString(TString *s)
+{
+    TString *r = string_createString(s->length);
+    for (size_t i = 0, e = s->length; e > 0; e--, i++) {
+        r->data[e-1] = s->data[i];
+    }
+    s = string_copyString(r);
```

Comment:
What is the purpose of copying this string here? Can't you simply `return r;`?

Reply:
The function was supposed to modify the string in-place.  I guess it would be better to return a whole different string pointer, not altering the original.

Reply:
Well, that's what it does now, it doesn't modify the passed string in-place.